### PR TITLE
Creata Data and Logic for balls

### PR DESCRIPTION
**Pull request by RadekBucki**

### CODE REVIEW

File: Data/Ball.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+namespace Data
+{
+    public class Ball
+    {
+        public int X { get; set; }
```
Comment (kambed) [inline Data/Ball.cs:5]: I think XPos and YPos gives more information and thats how its called in frontEnd

File: Logic/BallsManager.cs [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        }
+
+        public Ball CreateBallInRandomPlace()
+        {
+            Random r = new();
+            int x = r.Next(0, _boardWidth);
+            int y = r.Next(0, _boardHeight);
+            int xSpeed = r.Next(-100, 100);
+            int ySpeed = r.Next(-100, 100);
+            return CreateBall(x, y, xSpeed, ySpeed);
```
Comment (kambed) [inline Logic/BallsManager.cs:36]: Is it needed to save this to variables, couldn't that be inline? Worth thinking I don't know what's better

File: Logic/BallsManager.cs [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        }
+
+        public Ball[] GetAllBalls()
+        {
+            return _ballsRepository.GetBalls();
+        }
+
+        public void RemoveAllBalls()
+        {
+            _ballsRepository = new BallsRepository();
```
Comment (kambed) [inline Logic/BallsManager.cs:46]: I think this should be using method in data layer not creating new one.
Use _balls.clear() in BallsRepo

File: Logic/BallsManager.cs [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        }
+        
+        public Ball CreateBall(int x, int y, int xSpeed, int ySpeed)
+        {
+            if (x < 0 || x > _boardWidth || y < 0 || y > _boardHeight)
+            {
+                throw new ArgumentException("Coordinate out of board range.");
+            }
+            Ball ball = new(x, y, xSpeed, ySpeed);
+            _ballsRepository.Add(ball);
```
Comment (kambed) [inline Logic/BallsManager.cs:25]: This also could be inlined

File: Logic/BallsManager.cs [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+        public BallsManager(int boardWidth, int boardHeight)
+        {
+            _boardWidth = boardWidth;
+            _boardHeight = boardHeight;
+        }
+        
+        public Ball CreateBall(int x, int y, int xSpeed, int ySpeed)
+        {
+            if (x < 0 || x > _boardWidth || y < 0 || y > _boardHeight)
```
Comment (kambed) [inline Logic/BallsManager.cs:20]: Speed validation missing

File: Data/Ball.cs [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+        public int Y { get; set; }
```
Comment (kambed) [inline Data/Ball.cs:6]: I think that setters should be validated if you validate it in constructor

File: Tests/DataTests/BallTest.cs [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+            Assert.AreEqual(3, ball.Y);
+        }
+        [TestMethod]
+        public void SetBallSpeedTest()
+        {
+            ball.XSpeed = 2;
+            ball.YSpeed = 3;
+            Assert.AreEqual(2, ball.XSpeed);
+            Assert.AreEqual(3, ball.YSpeed);
+        }
```
Comment (kambed) [inline Tests/DataTests/BallTest.cs:33]: When setter validation added negative test are required

**Review by kambed — CHANGES_REQUESTED** [review]

File: Data/Ball.cs (hunk 6 above)
Comment (RadekBucki) [reply Data/Ball.cs:6]: There is  no validation in costructor. Validation is only in Logic Layer in `BallsManager` class

**Review by kambed — APPROVED** [review]

**Pull request merged**